selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/CreateStackTest.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.selenium.dashboard;
+
+import static org.eclipse.che.commons.lang.NameGenerator.generate;
+
+import com.google.inject.Inject;
+import org.eclipse.che.selenium.core.TestGroup;
+import org.eclipse.che.selenium.core.utils.WaitUtils;
+import org.eclipse.che.selenium.pageobject.dashboard.BuildStackFromRecipe;
+import org.eclipse.che.selenium.pageobject.dashboard.Dashboard;
+import org.eclipse.che.selenium.pageobject.dashboard.stacks.StackDetails;
+import org.eclipse.che.selenium.pageobject.dashboard.stacks.Stacks;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/** @author Skoryk Serhii */
+public class CreateStackTest {
+  private static final String COMPOSE_RECIPE =
+      "services:\n"
+          + " db:\n"
+          + "  image: eclipse/mysql\n"
+          + "environment:\n"
+          + "  MYSQL_ROOT_PASSWORD: password\n"
+          + "MYSQL_DATABASE: petclinic\n"
+          + "MYSQL_USER: petclinic\n"
+          + "MYSQL_PASSWORD: password\n"
+          + "\b\bmem_limit: 1073741824\n"
+          + "\b\bdev-machine:\n"
+          + "  image: eclipse/ubuntu_jdk8\n"
+          + "mem_limit: 2147483648\n"
+          + "depends_on:\n"
+          + "  - db";
+  private static final String DOCKERIMAGE_RECIPE = "eclipse/ubuntu_jdk8";
+  private static final String DOCKERFILE_RECIPE = "FROM eclipse/ubuntu_jdk8";
+  private static final String KUBERNETES_RECIPE =
+      "---\n"
+          + "kind: List\n"
+          + "items:\n"
+          + "-\n"
+          + "  apiVersion: v1\n"
+          + "kind: Pod\n"
+          + "metadata:\n"
+          + "  name: ws\n"
+          + "\b\bspec:\n"
+          + "  containers:\n"
+          + "  -\n"
+          + "  image: eclipse/ubuntu_jdk8\n"
+          + "name: dev-machine";
+
+  @Inject private Stacks stacks;
+  @Inject private Dashboard dashboard;
+  @Inject private StackDetails stackDetails;
+  @Inject private BuildStackFromRecipe buildStackFromRecipe;
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    dashboard.open();
+  }
+
+  @BeforeMethod
+  public void openStacksListPage() {
+    dashboard.waitDashboardToolbarTitle();
+    dashboard.selectStacksItemOnDashboard();
+    stacks.waitToolbarTitleName();
+
+    stacks.clickOnAddStackButton();
+    buildStackFromRecipe.waitCreateStackDialogVisible();
+  }
+
+  @AfterClass
+  public void deleteCreatedStacks() {
```

Comment:
This method should not throw an error if stacks didn't created form some reasons.